### PR TITLE
added doc for envis_preprocessor.py

### PR DESCRIPTION
**Pull request by bashirk**
## Description
This PR adds inline documentation to the envis_preprocessor.py file.

## Relevant Issue
Issue #269 

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/NimbleEdge/EnvisEdge/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/NimbleEdge/EnvisEdge/blob/main/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [NimbleEdge Styleguide](https://numpydoc.readthedocs.io/en/latest/format.html)

### CODE REVIEW

File: fedrec/user_modules/envis_preprocessor.py [hunk 1]
```diff
@@ -6,6 +6,42 @@
 
 @Registrable.register_class_ref
 class EnvisPreProcessor(Serializable):
+    """
+    This class preprocesses the data and stores it in the storage. It also
+    loads the data and the data description from the storage.
+
+    The data is stored in the storage as a dictionary, where the keys are
+    the splits and the values are the data.
+
```
Comment (divyanshugit) [inline fedrec/user_modules/envis_preprocessor.py:15]: Please elaborate a bit more. You can read about the classes that have been inherited in this file to give proper explanation and flow.

**Review by divyanshugit — CHANGES_REQUESTED** [review]

File: fedrec/user_modules/envis_preprocessor.py [hunk 2]
```diff
@@ -6,6 +6,44 @@
 
 @Registrable.register_class_ref
 class EnvisPreProcessor(Serializable):
+    """
+    The EnvisPreProcessor class is a wrapper around the dataset processor.
```
Comment (divyanshugit) [inline fedrec/user_modules/envis_preprocessor.py:10]: Can you please exclude this line and re-write the whole paragraph as in we don't have any dataset processor class in our codebase?

**Review by divyanshugit — CHANGES_REQUESTED** [review]

File: fedrec/user_modules/envis_preprocessor.py (hunk 2 above)
Comment (bashirk) [reply fedrec/user_modules/envis_preprocessor.py]: Thanks. I've re-written the paragraph.

File: fedrec/user_modules/envis_preprocessor.py [hunk 3]
```diff
@@ -6,6 +6,44 @@
...
 @Registrable.register_class_ref
 class EnvisPreProcessor(Serializable):
+    """
+    The EnvisPreProcessor class extends the Serializable class and is used
+    to preprocess the data before training. It preprocesses the data and
+    stores it in the storage before it is used by the model.
+
+    It uses the Serializable interface to serialize the data and store it
+    in the storage, where the data is later loaded by the model. It also
+    uses the Registratable interface to register the class in the registry.
```
Comment (divyanshugit) [inline fedrec/user_modules/envis_preprocessor.py:16]: Just rephrase the `It also
    uses the Registratable interface to register the class in the registry.` as `It adds the class to the registry using the Registrable interface.`

**Review by divyanshugit — CHANGES_REQUESTED** [review]

File: fedrec/user_modules/envis_preprocessor.py (hunk 3 above)
Comment (bashirk) [reply fedrec/user_modules/envis_preprocessor.py]: Done.

**Comment by divyanshugit** [comment]
LGTM!! @ramesht007

**Review by ramesht007 — APPROVED** [review]
LGTM!!

**Pull request merged**